spec.bs

```diff
@@ -257,14 +257,191 @@ dictionary AuctionAdConfig {
+  1. If |key| equals to "*", then set |auctionConfig|'s [=auction config/all buyers group limit=]
+    to |value|, and [=iteration/continue=].
+  1. Let |buyer| be parsing |key| to an [=url/origin=]. If it fails, [=exception/throw=] a
+    {{TypeError}}.
+  1. [=map/Set=] |auctionConfig|'s [=auction config/per buyer group limits=][|buyer|] to |value|.
+1. If |config|["{{AuctionAdConfig/perBuyerPrioritySignals}}"] [=map/exists=], [=map/for each=]
+  |key| → |value| of |config|["{{AuctionAdConfig/perBuyerPrioritySignals}}"]:
+  1. Let |signals| be an [=ordered map=] whose [=map/key|keys=] are [=strings=] and whose
+    [=map/value|values=] are double.
+  1. [=map/for each=] |k| → |v| of |value|:
+    1. If |k| [=string/starts with=] "browserSignals.", [=exception/throw=] a {{TypeError}}.
+    1. Set |signals|[|k|] to |v|.
+  1. If |key| equals to "*", then set |auctionConfig|'s
+    [=auction config/all buyers priority signals=] to |value|, and [=iteration/continue=].
+  1. Let |buyer| be parsing |key| to an [=url/origin=]. If it fails, [=exception/throw=] a
+    {{TypeError}}.
+  1. [=map/Set=] |auctionConfig|'s [=auction config/per buyer priority signals=][|buyer|] to
+    |signals|.
+1. If |config|["{{AuctionAdConfig/sellerTimeout}}"] [=map/exists=], set |auctionConfig|'s
+  [=auction config/seller timeout=] to |config|["{{AuctionAdConfig/sellerTimeout}}"].
+1. If |config|["{{AuctionAdConfig/componentAuctions}}"] [=map/exists=], [=list/for each=]
+  |component| in |config|["{{AuctionAdConfig/componentAuctions}}"]:
+  1. If |isTopLevel| is false, [=exception/throw=] a {{TypeError}}.
+  1. Let |componentAuction| be the result of running [=validate and convert auction ad config=] with
+    |component| and [=validate and convert auction ad config/isTopLevel=] set to false. If it fails,
+    return null.
+  1. [=list/Append=] |componentAuction| to |auctionConfig|'s [=auction config/component auctions=].
+1. If |config|["{{AuctionAdConfig/sellerExperimentGroupId}}"] [=map/exists=]:
+  1. Set |auctionConfig|'s [=auction config/seller experiment group id=] to
+    |config|["{{AuctionAdConfig/sellerExperimentGroupId}}"].
+1. Return |auctionConfig|.
+
+</div>
+
+To <dfn>start bidding and scoring</dfn> given an [=auction config=]
+|auctionConfig|:
+1. If |auctionConfig|'s [=auction config/component auctions=] are empty:
+  1. Let sellerWorklet be a worklet object. TODO: expand this.
+1. OtherWise:
+  1. [=list/For each=] |component| |auctionConfig|'s [=auction config/component auctions=]:
+    1. [=Start bidding and scoring=] with |component|.
+1. [=list/For each=] |buyer| in [=auction config/interest group buyers=]:
+  1. Let |bidderWorklet| be a worklet object for |buyer|. TODO: expand.
+  1. If |bidderWorklet| received, do [=fetch bidding javascript=] with |buyer|.
+  1. Otherwise, do OnBidderWorkletGenerateBidFatalError.
+  1. TODO: rewrite this part.
+
+To <dfn>fetch bidding javascript</dfn> with given |buyer|:

```

Comment:
Specify what kind of thing |buyer| is. You call this with a URL, but then you use a [=interest group/bidding url=] from it.

Reply:
yeah we have some internal data structures for a buyer's worklet to store needed information about that buyer, like the buyer's bidding url. Added a todo for that.